File: functions/alsa-lib/variants/c/2_muldiv_down_2.c
#include <stdint.h>

static inline int muldiv_down(int a, int b, int c)
{
	int64_t v = (int64_t)a * b / (int64_t)c;
	if (v > 2147483647LL) {
		return 2147483647;
	}
	if (v < -2147483648LL) {
		return -2147483648;
	}
	return (int)v;
}


int main(){
    muldiv_down(0,0,0);
}

File: functions/alsa-lib/variants/go/2_muldiv_down_2.go
package main

func muldiv_down(a int32, b int32, c int32) int32 {
	v := int64(a) * int64(b) / int64(c)
	if v > int64(2147483647) {
		return int32(2147483647)
	}
	if v < int64(-2147483648) {
		return int32(-2147483648)
	}
	return int32(v)
}


func main(){
    muldiv_down(0,0,0)
}